pkg/deps/config/suppliers.go

```diff
@@ -143,17 +149,23 @@ func NewSupplyQueryClientContextFn(pocketQueryNodeURL string) SupplierFn {
 // NewSupplyTxClientContextFn returns a function with constructs a ClientContext
 // instance with the given cmd and returns a new depinject.Config which is
 // supplied with the given deps and the new ClientContext.
-func NewSupplyTxClientContextFn(pocketTxNodeURL string) SupplierFn {
+func NewSupplyTxClientContextFn(txNodeGRPCURL *url.URL) SupplierFn {
 	return func(_ context.Context,
 		deps depinject.Config,
 		cmd *cobra.Command,
 	) (depinject.Config, error) {
 		// Temporarily store the flag's current value
-		tmp := cosmosflags.FlagNode
+		// TODO_TECHDEBT(#223) Retrieve value from viper instead, once integrated.
+		tmpGRPC, err := cmd.Flags().GetString(cosmosflags.FlagGRPC)
```

Comment:
Just confirming, the only remaining reason that we MUST use `cosmosflags.FlagGRPC` and care about resetting its value, is for the benefit of the `TxClient`, which is still using `sdkclient.Context`, right (although it can now be refactored, not to)?

Reply:
Maybe it's worth adding a `// TODO_TECHDEBT(#256): ...` comment.

Reply:
Left some comments on this as well. If we can get rid of `tmpGRPC` in a followup PR, that'd be 🔥